estimagic/differentiation/differentiation_auxiliary.py

```diff
@@ -1,68 +1,63 @@
 """Auxiliary functions for a one step differentiation."""
 
 
-def central(func, f_x0, params, var, h, *func_args, **func_kwargs):
+def central(internal_func, f_x0, params_value, i, h):
     """Calculate the central difference.
 
     Args:
-        func (func): A function that maps params_sr into a float.
-        f_x0 (func output): The function value at params_sr, func_args and func_kwargs.
-        params (DataFrame): see :ref:`parmas_df`
-        var (string): The variable of which the finite difference is calculated.
+        internal_func (func): A function wrapping the original function, s.t. it is
+        only dependent on the variable to differentiate.
+        f_x0 (func output): The function value at params_value.
```

Comment:
func_output is not a Python type. What types are allowed? I didn't see this last time. Maybe you have to change it at several places.

Reply:
Changed type to numpy array-